benchmark/external_benchmarks.jl

```diff
@@ -0,0 +1,164 @@
+# ==========================================================================
+# This script compares the facilities offered by CovarianceEstimation.jl
+# to existing packages and compares the performances for a set of reasonably
+# large matrices.
+#
+# Libraries considered:
+#
+# - [python] sklearn
+# - [R] corpcor
+# - [Matlab/Octave] Ledoit & Wolfe's implementations
+#
+# ==========================================================================
+
+using CovarianceEstimation
+using Random:seed!
+using DelimitedFiles
+using Statistics
+using LinearAlgebra
+
+seed!(65489) # random seed for reproducibility
+
+p = [20, 40, 200, 400]
+n = [40, 20, 400, 200]
+L = [randn(pᵢ, pᵢ) for pᵢ ∈ p]
+C = [Lᵢ' * Lᵢ for Lᵢ ∈ L]
+
+X = [randn(n[i], p[i]) * L[i] for i ∈ eachindex(p)]
+
+for Cᵢ ∈ C
+    pᵢ = size(Cᵢ, 1)
+    writedlm("benchmark/tmpmat/C_$(pᵢ).csv", Cᵢ)
+end
+
+for Xᵢ ∈ X
+    nᵢ, pᵢ = size(Xᵢ)
+    writedlm("benchmark/tmpmat/X_$(nᵢ)x$(pᵢ).csv", Xᵢ)
+    # octave is unstable for eigenvalue decomposition so we do everything here
+    F = eigen(Xᵢ' * Xᵢ / size(Xᵢ, 1))
+    writedlm("benchmark/tmpmat/S_lambda_$(nᵢ)x$(pᵢ).csv", F.values)
+    writedlm("benchmark/tmpmat/S_u_$(nᵢ)x$(pᵢ).csv", F.vectors)
+end
+
+# =========
+# SKLEARN
+# =========
+oas = LinearShrinkageEstimator(
+        target=DiagonalCommonVariance(),
+        shrinkage=:oas)
+lw = LinearShrinkageEstimator(
+        target=DiagonalCommonVariance(),
+        shrinkage=:lw)
+times = zeros(length(p))
+res = zeros(length(p))
+for i ∈ eachindex(p)
+    Xᵢ = X[i]
+    Cᵢ = C[i]
+    start = time()
+    C_oas = cov(Xᵢ, oas)
+    times[i] = time() - start
+    res[i] = norm(C_oas - Cᵢ)
+end
+res_ours = res
+times_ours = times
+res_sklearn = [  51.05167435  214.21678806 1602.04944525 6533.94732887][:]
+times_sklearn = [0.00026703 0.00023913 0.00225997 0.00262403][:]
+
+δ = mean(res_ours - res_sklearn)
+speedup = mean(times_sklearn ./ times_ours)
+
+for i ∈ eachindex(p)
+    Xᵢ = X[i]
+    Cᵢ = C[i]
+    start = time()
```

Comment:
ok so I actually didn't do this on purpose because I thought it was a bit of an unfair comparison, a user would not use the function multiple time but rather just the one time and therefore would experience the compile time. I effectively do a rough "pre-compile" by running the function for all estimators on small data before though.

Reply:
It's not totally fair either way but we can leave it this way for now.